### ISSUE TITLE
[Bug]: panic: internal error: mpool freed more bytes than alloc

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#5491]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):0.6.0
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

 1970, 19700101, 1970-11-11 22:22:22) Error: truncated type DATE value 19700101 for column f3, 1
github.com/matrixorigin/matrixone/pkg/frontend.logStatementStringStatus
        /data1/sudong/mo-regression/matrixone/pkg/frontend/util.go:454
github.com/matrixorigin/matrixone/pkg/frontend.logStatementStatus
        /data1/sudong/mo-regression/matrixone/pkg/frontend/util.go:446
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).doComQuery
        /data1/sudong/mo-regression/matrixone/pkg/frontend/mysql_cmd_executor.go:2436
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).ExecRequest
        /data1/sudong/mo-regression/matrixone/pkg/frontend/mysql_cmd_executor.go:2505
github.com/matrixorigin/matrixone/pkg/frontend.(*Routine).Loop
        /data1/sudong/mo-regression/matrixone/pkg/frontend/routine.go:125
panic: internal error: mpool freed more bytes than alloc

goroutine 44523615 [running]:
github.com/matrixorigin/matrixone/pkg/common/mpool.(*MPoolStats).RecordFree(0xc530b3f040?, 0xc976de5bb8?)
        /data1/sudong/mo-regression/matrixone/pkg/common/mpool/mpool.go:93 +0x74
github.com/matrixorigin/matrixone/pkg/common/mpool.(*MPool).Free(0xdf43e1ce10?, {0xd08c843010, 0xc8ae326cb8?, 0xdf43e1c780?})
        /data1/sudong/mo-regression/matrixone/pkg/common/mpool/mpool.go:456 +0x85
github.com/matrixorigin/matrixone/pkg/container/vector.(*Vector).Free(0xc3211dc080, 0xc8ae326c60?)
        /data1/sudong/mo-regression/matrixone/pkg/container/vector/vector.go:634 +0x4c
github.com/matrixorigin/matrixone/pkg/container/batch.(*Batch).Clean(0xc3211dc000, 0xdd0b311180)
        /data1/sudong/mo-regression/matrixone/pkg/container/batch/batch.go:241 +0x75
github.com/matrixorigin/matrixone/pkg/vm/pipeline.(*Pipeline).MergeRun.func1()
        /data1/sudong/mo-regression/matrixone/pkg/vm/pipeline/pipeline.go:130 +0xcd
github.com/matrixorigin/matrixone/pkg/vm/pipeline.(*Pipeline).MergeRun(0xc976de5d38, 0xc8ab00a3c0)
        /data1/sudong/mo-regression/matrixone/pkg/vm/pipeline/pipeline.go:148 +0x1ce
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Scope).MergeRun(0xc395716d20, 0xdd2b8c7e10)
        /data1/sudong/mo-regression/matrixone/pkg/sql/compile/scope.go:110 +0x617
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Scope).JoinRun(0xc395716d20, 0xdd2b8c7e10)
        /data1/sudong/mo-regression/matrixone/pkg/sql/compile/scope.go:225 +0x505
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Scope).ParallelRun(0xc397486fa0?, 0x417020?)

**Comment by m-schen** [comment]
I have fixed this bug. please have a try.

**Comment by aressu1985** [comment]
fixed: commit fc685762b89cd931bdac2c09b93f6bab4ceeb6d5
